pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -231,26 +229,68 @@ func TestDatabaseLevelChangefeedWithIncludeFilter(t *testing.T) {
 
 	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
-		expectSuccess := func(stmt string) {
-			successfulFeed := feed(t, f, stmt)
-			defer closeFeed(t, successfulFeed)
-			_, err := successfulFeed.Next()
-			require.NoError(t, err)
-		}
 		sqlDB := sqlutils.MakeSQLRunner(s.DB)
-		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
-		sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'initial')`)
-		sqlDB.Exec(t, `UPSERT INTO foo VALUES (0, 'updated')`)
-		sqlDB.Exec(t, `CREATE TABLE foo2 (a INT PRIMARY KEY, b STRING)`)
-		sqlDB.Exec(t, `INSERT INTO foo2 VALUES (0, 'initial')`)
-		sqlDB.Exec(t, `UPSERT INTO foo2 VALUES (0, 'updated')`)
+		for i := range 4 {
+			name := fmt.Sprintf("foo%d", i+1)
+			sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE %s (a INT PRIMARY KEY, b STRING)`, name))
+			sqlDB.Exec(t, fmt.Sprintf(`INSERT INTO %s VALUES (0, 'initial')`, name))
+			sqlDB.Exec(t, fmt.Sprintf(`UPSERT INTO %s VALUES (0, 'updated')`, name))
+		}
+
+		sqlDB.Exec(t, `CREATE SCHEMA private`)
+		sqlDB.Exec(t, `CREATE TABLE private.foo1 (a INT PRIMARY KEY, b STRING)`)
+		sqlDB.Exec(t, `INSERT INTO private.foo1 VALUES (0, 'initial')`)
+		// Test that if there are multiple tables with the same name the correct
+		// one will still be found using the default schema of public.
+		feed1 := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES foo1`)
+		defer closeFeed(t, feed1)
+		assertPayloads(t, feed1, []string{
+			`foo1: [0]->{"after": {"a": 0, "b": "updated"}}`,
+		})
+
+		// Test that we can include multiple tables.
+		feed2 := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES foo1,foo2`)
+		defer closeFeed(t, feed2)
+		assertPayloads(t, feed2, []string{
+			`foo1: [0]->{"after": {"a": 0, "b": "updated"}}`,
+			`foo2: [0]->{"after": {"a": 0, "b": "updated"}}`,
+		})
+
+		// Test that we can handle fully qualfied, partially qualified, and unqualified table names.
+		feed3 := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES d.public.foo2, public.foo3, foo4`)
+		defer closeFeed(t, feed3)
+		assertPayloads(t, feed3, []string{
+			`foo2: [0]->{"after": {"a": 0, "b": "updated"}}`,
+			`foo3: [0]->{"after": {"a": 0, "b": "updated"}}`,
+			`foo4: [0]->{"after": {"a": 0, "b": "updated"}}`,
+		})
 
-		expectSuccess(`CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES foo`)
-		expectSuccess(`CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES foo,foo2`)
-		expectSuccess(`CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES d.bar.fizz, foo.foo2, foo`)
+		// Test that we can handle tables that don't exist.
+		feed4 := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES foo1,bob`)
+		defer closeFeed(t, feed4)
+		assertPayloads(t, feed4, []string{
+			`foo1: [0]->{"after": {"a": 0, "b": "updated"}}`,
+		})
+
+		// Test that fully qualified table names outside of the target database will
+		// cause an error.
+		expectErrCreatingFeed(t, f, `CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES fizz.buzz.foo`,
```

Comment:
Could you add one more error case for `d.foo1`?

Reply:
Maybe add a test that it doesn't error with `d.foo1` but also doesn't get any rows for `d.public.foo1` then? You might have to make the changefeed also watch a different table to make sure there's at least some messages